core/vm/gas_table.go

```diff
@@ -116,79 +116,70 @@ func gasReturnDataCopy(gt params.GasTable, evm *EVM, contract *Contract, stack *
 }
 
 func gasSStore(gt params.GasTable, evm *EVM, contract *Contract, stack *Stack, mem *Memory, memorySize uint64) (uint64, error) {
-	var (
-		y, x = stack.Back(1), stack.Back(0)
-		val  = evm.StateDB.GetState(contract.Address(), common.BigToHash(x))
-	)
-	// This checks for 3 scenario's and calculates gas accordingly
-	// 1. From a zero-value address to a non-zero value         (NEW VALUE)
-	// 2. From a non-zero value address to a zero-value address (DELETE)
-	// 3. From a non-zero to a non-zero                         (CHANGE)
-	if val == (common.Hash{}) && y.Sign() != 0 {
-		// 0 => non 0
-		return params.SstoreSetGas, nil
-	} else if val != (common.Hash{}) && y.Sign() == 0 {
-		// non 0 => 0
-		evm.StateDB.AddRefund(params.SstoreRefundGas)
-		return params.SstoreClearGas, nil
-	} else {
-		// non 0 => non 0 (or 0 => 0)
-		return params.SstoreResetGas, nil
-	}
-}
-
-// gasSStoreEip1283 calculates SSTORE gas cost according to EIP-1283
-func gasSStoreEip1283(gt params.GasTable, evm *EVM, contract *Contract, stack *Stack, mem *Memory, memorySize uint64) (uint64, error) {
 	var (
 		y, x    = stack.Back(1), stack.Back(0)
 		current = evm.StateDB.GetState(contract.Address(), common.BigToHash(x))
 	)
-	//1. If current value equals new value (this is a no-op), 200 gas is deducted.
-	//2. If current value does not equal new value
-	//  2.1 If original value equals current value (this storage slot has not been changed by the current execution context)
-	//    2.1.1 If original value is 0, 20000 gas is deducted.
-	//	  2.1.2 Otherwise, 5000 gas is deducted. If new value is 0, add 15000 gas to refund counter.
-	//	2.2 If original value does not equal current value (this storage slot is dirty), 200 gas is deducted. Apply both of the following clauses.
-	//	  2.2.1 If original value is not 0
-	//      2.2.1.1 If current value is 0 (also means that new value is not 0), remove 15000 gas from refund counter. We can prove that refund counter will never go below 0.
-	//      2.2.1.2 If new value is 0 (also means that current value is not 0), add 15000 gas to refund counter.
-	//	  2.2.2 If original value equals new value (this storage slot is reset)
-	//      2.2.2.1 If original value is 0, add 19800 gas to refund counter.
-	//	    2.2.2.2 Otherwise, add 4800 gas to refund counter.
-	new := common.BigToHash(y)
-	if current == new {
-		// 1. current == new
-		return 200, nil
-	}
-	original := evm.StateDB.GetStateOriginal(contract.Address(), common.BigToHash(x))
-	// 2
-	if original == current { // 2.1
-		if original == (common.Hash{}) { // 2.1.1
-			return 20000, nil
+	// The legacy gas metering only takes into consideration the current state
+	if !evm.chainRules.IsConstantinople {
```

Comment:
I personally think it was cleaner to have two separate gas calc functions, one enabled in Constantinople and one earlier. That makes the respective methods smaller, and saves this check on every `SSTORE`.

Reply:
It's a single bool comparison vs. an sstore potentially loads of data from disk. Other gas calculations (e.g. `gasSuicide`) also use the same logic of having a single method per opcode and potentially checking whether a fork is enabled or not. It feels a bit cleaner for me to have a single method, but if you really disagree we can split it.

Reply:
Yeah, it's not a heavy toll, the main reason for me was more that it allowed splitting the methods into two separate units. 
I don't feel strongly about it, you're the main maintainer :)